drivers/gpio_exp/include/gpio_exp_reg.h

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright (C) 2018 Acutam Automation, LLC
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     drivers_gpio_exp
+ *
+ * @{
+ *
+ * @file
+ * @brief       GPIO expansion default registry
```

Comment:
I don't quite like that gpio_exp comes with its own registry. Is this actually necessary?

Reply:
Yes. Expanders are addressed by a number that is used with ```GPIO_EXP_PIN(foo_dev, bar_pin)```. The number ```foo_dev``` cannot be known until linking the binary, because the board may have more than one GPIO expander, so ```foo_dev``` cannot be a macro or enum set by the driver. Also, order is important, so automatically building the registry with a linked list cannot be relied on, because changes to code may change the order of device numbers. This leaves a manually defined array as the remaining satisfactory choice, which would be set up as part of the board includes by the developer porting the board.

Reply:
Note: I think I was using the word "registry" too loosely here. It is not a registry in the sense of e.g. SAUL. I will change the name of the file to ```gpio_exp_conf.h```

Reply:
Should be clearer now.